shell/platform/fuchsia/flutter/accessibility_bridge.cc

```diff
@@ -927,4 +885,68 @@ void AccessibilityBridge::FillInspectTree(int32_t flutter_node_id,
 }
 #endif  // !FLUTTER_RELEASE
 
+AccessibilityBridge::FuchsiaAtomicUpdate::FuchsiaAtomicUpdate(
+    std::queue<FuchsiaAtomicUpdate>* atomic_updates)
+    : atomic_updates_(atomic_updates) {}
+
+void AccessibilityBridge::FuchsiaAtomicUpdate::AddNodeUpdate(
+    fuchsia::accessibility::semantics::Node node,
+    size_t size) {
+  if (size > kMaxMessageSize) {
+    // TODO(MI4-2531, FIDL-718): Remove this
+    // This is defensive. If, despite our best efforts, we ended up with a node
+    // that is larger than the max fidl size, we send no updates.
+    PrintNodeSizeError(node.node_id());
+    return;
+  }
+  updates_.emplace_back(std::move(node), size);
+}
+
+void AccessibilityBridge::FuchsiaAtomicUpdate::AddNodeDeletion(uint32_t id) {
+  deletions_.push_back(id);
+}
+
+void AccessibilityBridge::FuchsiaAtomicUpdate::Apply(
```

Comment:
Why cant we make Apply a member of AccessibilityBridge instead of a member of FuchsiaAtomicUpdate?  The lifetime is confusing and prone to bugs; with atomic_updates_ there is a circular dependency between AccessibilityBridge and FuchsiaAtomicUpdate

Reply:
I think I fixed your main concern in this new patch. Let me know what you think.